File: C++/freecodecamp_c++_course/14_building_a_better_calculator.cpp
#include <iostream>
#include <math.h>

using namespace std;

double calculate()
{
  double num1, num2;
  char operation;

  cout << "Enter a number: ";
  cin >> num1;

  cout << "Enter an operation (+, -, *, /, or \%): ";
  cin >> operation;

  cout << "Enter another number: ";
  cin >> num2;

  switch (operation) {
    case '+':
      return num1 + num2;
    case '-':
      return num1 - num2;
    case '*':
      return num1 * num2;
    case '/':
      return num1 / num2;
    case '%':
      return fmod(num1, num2);
    default:
      return 0;
  }
}

int main()
{
  cout << calculate() << endl;
  
  return 0;
}

File: C#/freecodecamp_c#_course/14_building_a_better_calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuildingABetterCalculator
{
  class Program
  {
    static void Main(string[] args)
    {
      Console.Write("Enter a number: ");
      double num1 = Convert.ToDouble(Console.ReadLine());

      Console.Write("Enter an operation: ");
      char operatorChar = Convert.ToChar(Console.ReadLine());

      Console.Write("Enter another number: ");
      double num2 = Convert.ToDouble(Console.ReadLine());

      switch (operatorChar)
      {
        case '+':
          Console.WriteLine(num1 + num2);
          break;
        case '-':
          Console.WriteLine(num1 - num2);
          break;
        case '*':
          Console.WriteLine(num1 * num2);
          break;
        case '/':
          Console.WriteLine(num1 / num2);
          break;
        case '%':
          Console.WriteLine(num1 % num2);
          break;
        default:
          Console.WriteLine("Invalid operation");
          break;
      }
    }
  }
}
